arcgis-ios-sdk-samples/Layers/Display KML/DisplayKMLViewController.swift

```diff
@@ -0,0 +1,186 @@
+            sourceTableViewController.selectionHandler = { [weak self] (dataSource) in
+                guard let self = self else{
+                    return
+                }
+                // close the popover
+                self.presentedViewController?.dismiss(animated: true)
+                // set the displayed source based on the selection
+                self.displayedSource = dataSource
+            }
+        }
+    }
+    
+    //MARK: - KML Loading
+    
+    private func display(kmlLayer: AGSKMLLayer) {
+        
+        // Clear the existing layers from the map
+        mapView.map?.operationalLayers.removeAllObjects()
+        // Add the KML layer to the map
+        mapView.map?.operationalLayers.add(kmlLayer)
+        
+        // Show the progress indicator
+        SVProgressHUD.show(withStatus: "Loading KML Layer")
+        
+        // This load call is not required, but it allows for error
+        // feedback and progress indication
+        kmlLayer.load {[weak self] (error) in
+            
+            // Close the progress indicator
+            SVProgressHUD.dismiss()
+
+            if let self = self,
+                let error = error {
+                // Display the error if one occurred
+                let alertController = UIAlertController(title: error.localizedDescription, message: nil, preferredStyle: .alert)
+                alertController.addAction(UIAlertAction(title: "OK", style: .cancel))
+                self.present(alertController, animated: true, completion: nil)
+            }
+        }
+        
+    }
+    
+    private func changeSourceToURL() {
+        /// A URL of a remote KML file
+        let kmlDatasetURL = URL(string: "https://www.wpc.ncep.noaa.gov/kml/noaa_chart/WPC_Day1_SigWx.kml")!
+        let kmlDataset = AGSKMLDataset(url: kmlDatasetURL)
+        /// A KML layer created from a remote KML file
+        let kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+        display(kmlLayer: kmlLayer)
+    }
+    
+    private func changeSourceToLocalFile() {
+        /// A dataset created by referencing the name of a KML file in the app bundle
+        let kmlDataset = AGSKMLDataset(name: "US_State_Capitals")
+        /// A KML layer created from a local KML file
+        let kmlLayer = AGSKMLLayer(kmlDataset: kmlDataset)
+        display(kmlLayer: kmlLayer)
+    }
+    
+    private func changeSourceToPortalItem() {
+        let portal = AGSPortal.arcGISOnline(withLoginRequired: false)
+        /// A remote KML portal item
+        let portalItem = AGSPortalItem(portal: portal, itemID: "9fe0b1bfdcd64c83bd77ea0452c76253")
+        /// A KML layer created from an ArcGIS Online portal item
+        let kmlLayer = AGSKMLLayer(item: portalItem)
+        display(kmlLayer: kmlLayer)
+    }
+    
+}
+
+extension DisplayKMLViewController: UIPopoverPresentationControllerDelegate {
+    
+    func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
```

Comment:
Per the documentation, please use `adaptivePresentationStyle(for:traitCollection:)` instead.

Reply:
Thanks for the catch! I didn't realize there were two methods.